presentation/components/src/test/instance-filter-builder/InstanceFilterConverter.test.ts

```diff
@@ -0,0 +1,477 @@
+      targetClassInfo: classBInfo,
+      relationshipInfo: classBToCInfo,
+      isForwardRelationship: false,
+      isPolymorphicRelationship: true,
+      isPolymorphicTargetClass: true,
+    }];
+    const propertyInfo = createTestPropertyInfo({ classInfo: classCInfo });
+    const classCPropertiesField = createTestPropertiesContentField({ properties: [{ property: propertyInfo }] });
+    const classCNestedField = createTestNestedContentField({ nestedFields: [classCPropertiesField], pathToPrimaryClass: pathCToB });
+    const classBNestedField = createTestNestedContentField({ nestedFields: [classCNestedField], pathToPrimaryClass: pathBToA });
+
+    beforeEach(() => {
+      classCPropertiesField.resetParentship();
+      classCNestedField.resetParentship();
+      classBNestedField.resetParentship();
+    });
+
+    it("in single condition", async () => {
+      classCPropertiesField.rebuildParentship(classCNestedField);
+      const filter: PresentationInstanceFilterCondition = {
+        field: classCPropertiesField,
+        operator: PropertyFilterRuleOperator.IsNull,
+      };
+      const { expression, relatedInstances } = await convertToInstanceFilterDefinition(filter, testImodel);
+      expect(expression).to.be.eq(`${createAlias("C")}.${propertyInfo.name} IS NULL`);
+      expect(relatedInstances).to.be.lengthOf(1).and.containSubset([{
+        pathFromSelectToPropertyClass: [{
+          sourceClassName: classBInfo.name,
+          targetClassName: classCInfo.name,
+          relationshipName: classBToCInfo.name,
+          isForwardRelationship: true,
+        }],
+        alias: createAlias("C"),
+      }]);
+    });
+
+    it("in multiple conditions", async () => {
+      classCPropertiesField.rebuildParentship(classCNestedField);
+      const filter: PresentationInstanceFilterConditionGroup = {
+        operator: PropertyFilterRuleGroupOperator.And,
+        conditions: [{
+          field: classCPropertiesField,
+          operator: PropertyFilterRuleOperator.IsNull,
+        }, {
+          field: classCPropertiesField,
+          operator: PropertyFilterRuleOperator.IsNotNull,
+        }],
+      };
+      const { expression, relatedInstances } = await convertToInstanceFilterDefinition(filter, testImodel);
+      expect(expression).to.be.eq(`(${createAlias("C")}.${propertyInfo.name} IS NULL AND ${createAlias("C")}.${propertyInfo.name} IS NOT NULL)`);
+      expect(relatedInstances).to.be.lengthOf(1).and.containSubset([{
+        pathFromSelectToPropertyClass: [{
+          sourceClassName: classBInfo.name,
+          targetClassName: classCInfo.name,
+          relationshipName: classBToCInfo.name,
+          isForwardRelationship: true,
+        }],
+        alias: createAlias("C"),
+      }]);
+    });
+
+    it("in deeply nested condition field", async () => {
+      classCPropertiesField.rebuildParentship(classCNestedField);
+      classCNestedField.rebuildParentship(classBNestedField);
```

Comment:
Calling `classCNestedField.rebuildParentship(classBNestedField)` should be enough - it traverses into nested fields and sets their parents